homeassistant/components/android_ip_webcam/sensor.py

```diff
@@ -1,75 +1,79 @@
+        [IPWebcamSensor(coordinator, description) for description in sensor_types]
+    )
 
-    all_sensors = []
 
-    for sensor in sensors:
-        all_sensors.append(IPWebcamSensor(name, host, ipcam, sensor))
-
-    async_add_entities(all_sensors, True)
-
-
-class IPWebcamSensor(AndroidIPCamEntity, SensorEntity):
+class IPWebcamSensor(AndroidIPCamBaseEntity, SensorEntity):
     """Representation of a IP Webcam sensor."""
 
-    def __init__(self, name, host, ipcam, sensor):
-        """Initialize the sensor."""
-        super().__init__(host, ipcam)
+    _attr_has_entity_name = True
 
-        self._sensor = sensor
-        self._mapped_name = KEY_MAP.get(self._sensor, self._sensor)
-        self._attr_name = f"{name} {self._mapped_name}"
+    def __init__(
+        self,
+        coordinator: AndroidIPCamDataUpdateCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{coordinator.config_entry.entry_id}-{description.key}"
         self._attr_native_value = None
         self._attr_native_unit_of_measurement = None
+        self.entity_description = description
 
-    async def async_update(self):
-        """Retrieve latest state."""
-        if self._sensor in ("audio_connections", "video_connections"):
-            if not self._ipcam.status_data:
-                return
-            self._attr_native_value = self._ipcam.status_data.get(self._sensor)
-            self._attr_native_unit_of_measurement = "Connections"
-        else:
-            (
-                self._attr_native_value,
-                self._attr_native_unit_of_measurement,
-            ) = self._ipcam.export_sensor(self._sensor)
+    @property
+    def native_value(self) -> StateType:
+        """Return sensor native value."""
+        if self.entity_description.key in ("audio_connections", "video_connections"):
+            return self._ipcam.status_data.get(self.entity_description.key)
+
+        value, _ = self._ipcam.export_sensor(self.entity_description.key)
+        return value
+
+    @property
+    def native_unit_of_measurement(self) -> str | None:
+        """Return sensor native unit of measurement."""
+        _, unit = self._ipcam.export_sensor(self.entity_description.key)
+        return unit
 
     @property
-    def icon(self):
+    def icon(self) -> str | None:
         """Return the icon for the sensor."""
-        if self._sensor == "battery_level" and self._attr_native_value is not None:
-            return icon_for_battery_level(int(self._attr_native_value))
-        return ICON_MAP.get(self._sensor, "mdi:eye")
+        if self.entity_description.key == "battery_level":
+            battery_level = (
+                int(self.native_value)
+                if self.native_value is not None
+                else self.native_value
+            )
+            return icon_for_battery_level(battery_level)
```

Comment:
We shouldn't use `icon_for_battery_level`. Just set battery device class on the appropriate description instead.